system/Cache/Handlers/PredisHandler.php

```diff
@@ -71,31 +129,11 @@ public function get(string $key)
     {
         $key = static::validateKey($key);
 
-        $data = array_combine(
-            ['__ci_type', '__ci_value'],
-            $this->redis->hmget($key, ['__ci_type', '__ci_value'])
-        );
-
-        if (! isset($data['__ci_type'], $data['__ci_value']) || $data['__ci_value'] === false) {
-            return null;
+        if(!(is_null($data = $this->redis->get($key)))) {
+            $data = unserialize($data);
```

Comment:
Storing php-serialized data is faster and more reliable than hard-coding the data types we anticipate.